k8s-charm/lib/charms/juju_dashboard/v0/juju_dashboard.py

```diff
@@ -78,10 +80,12 @@ def __init__(self, data):
         """
         # FIXME: Quick hack to fix a k8s bug in the controller charm.
         controller_url = data.get("controller-url")
-        controller_url = controller_url.replace("[", "").replace("]", "")
+        controller_url = (
+            controller_url.replace("[", "").replace(":0]", "").replace("]", "")
+        )
         # End quick hack!
         self._data = {
-            "controller_url": re.sub(r'\/api$', '', controller_url),
+            "controller_url": re.sub(r"\/api$", "", controller_url),
```

Comment:
If I'm reading this right, we're removing `/api` from the provided URL? If so, this is something we should also fix on the controller charm side.